compiler/vmgen.nim

```diff
@@ -986,11 +986,11 @@ proc genMagic(c: PCtx; n: PNode; dest: var TDest; m: TMagic) =
     c.freeTemp(tmp)
     c.freeTemp(tmp2)
 
-  of mShlI: genBinaryABCnarrowU(c, n, dest, opcShlInt)
+  of mShlI: genBinaryABCnarrow(c, n, dest, opcShlInt)
   of mAshrI: genBinaryABCnarrow(c, n, dest, opcAshrInt)
-  of mBitandI: genBinaryABCnarrowU(c, n, dest, opcBitandInt)
-  of mBitorI: genBinaryABCnarrowU(c, n, dest, opcBitorInt)
-  of mBitxorI: genBinaryABCnarrowU(c, n, dest, opcBitxorInt)
+  of mBitandI: genBinaryABCnarrow(c, n, dest, opcBitandInt)
+  of mBitorI: genBinaryABCnarrow(c, n, dest, opcBitorInt)
+  of mBitxorI: genBinaryABCnarrow(c, n, dest, opcBitxorInt)
```

Comment:
you can leave out the narrowing step entirely. `and`, `or` and `xor` can only reduce the total amount of set bits.

Reply:
done